web/struct_tree_layer_builder.js

```diff
@@ -104,7 +113,23 @@ class StructTreeLayerBuilder {
   #setAttributes(structElement, htmlElement) {
     const { alt, id, lang } = structElement;
     if (alt !== undefined) {
-      htmlElement.setAttribute("aria-label", removeNullCharacters(alt));
+      // Don't add the label in the struct tree layer but on the annotation
+      // in the annotation layer.
+      let added = false;
+      for (const child of structElement.children) {
+        if (child.type === "annotation") {
+          let attrs = this.#elementAttributes.get(child.id);
+          if (!attrs) {
+            attrs = Object.create(null);
+            this.#elementAttributes.set(child.id, attrs);
+          }
+          attrs["aria-label"] = removeNullCharacters(alt);
```

Comment:
Could multiple children have the same alt-text?
If so we can perhaps add a `const altText = removeNullCharacters(alt);` variable *above* the loop, and use that one both here in the the `setAttribute`-call below.